tests/llms/test_cohere.py

```diff
@@ -0,0 +1,149 @@
+    )
+
+
+def mock_chat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+async def mock_achat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+@pytest.mark.skipif(cohere is None, reason="cohere not installed")
+def test_completion_model_basic(monkeypatch: MonkeyPatch) -> None:
+    monkeypatch.setattr(
+        "llama_index.llms.cohere.completion_with_retry", mock_completion_with_retry
+    )
+    mock_api_key = "5secFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+    llm = Cohere(model="command", api_key=mock_api_key)
+    test_prompt = "test prompt"
+    response = llm.complete(test_prompt)
+    assert response.text == "\n\nThis is indeed a test"
+
+    monkeypatch.setattr(
+        "llama_index.llms.cohere.completion_with_retry", mock_chat_with_retry
+    )
+
+    message = ChatMessage(role="user", content=test_prompt)
+    chat_response = llm.chat([message])
+    assert chat_response.message.content == "\n\nThis is indeed a test"
+
+
+@pytest.mark.skipif(cohere is None, reason="cohere not installed")
+@pytest.mark.asyncio()
```

Comment:
Out of curiosity, why do you have this `asyncio` decorator here?

Reply:
That test method is async. My understanding from reading other llms tests is that async test functions need the pytest.mark.asyncio decorator.

Reply:
Okay, disclaimer I am not great with `asyncio`. I think you're not using the `event_loop` the decorator creates (e.g. see https://stackoverflow.com/a/57463137), so you may not need this decorator

Reply:
I tested it without the decorator and it throws :
```PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.```
In the stackoverflow post you sent, the event_loop isn't created by the decorator, its a normal arguement to the function. The decorator there is used because the function itself is async and uses await in the body. So I think the decorator should be kept.

Reply:
Thanks for circling back on this!  Appreciated